src/deploy/functions/remoteSource.ts

```diff
@@ -0,0 +1,135 @@
+import * as fs from "fs";
+import * as path from "path";
+
+import { URL } from "url";
+
+import { FirebaseError } from "../../error";
+import { logger } from "../../logger";
+import { logLabeledBullet, resolveWithin } from "../../utils";
+import { dirExistsSync, fileExistsSync } from "../../fsutils";
+import * as downloadUtils from "../../downloadUtils";
+import * as unzipModule from "../../unzip";
+
+/**
+ * Downloads a remote source to a temporary directory and returns the absolute path
+ * to the source directory. 
+ * 
+ * @param repository Remote URL (e.g. https://github.com/org/repo) or shorthand (org/repo)
+ * @param ref Git ref to fetch (tag/branch/commit)
+ * @param subDir Optional subdirectory within the repo to use
```

Comment:
`destDir` is missing from jsdoc. Is that the spot to put the fetched source code?

Reply:
good catch!